app/scripts/services/authorization.js

```diff
@@ -0,0 +1,169 @@
+            self.canCreateSomeResource = true;
+            self.wasCanCreateSomeResourceSet = true;
+            pageRules.createSomeResource = true;
+          }
+        });
+      } else {
+        // If there are no page rules to check, just use previously set 'wasCanCreateSomeResourceSet' attribute 
+        // whether user can create some resources. If the attribute isn't cached set it by going through all the
+        // rules. 
+        if (self.wasCanCreateSomeResourceSet) {
+          pageRules = {
+            createSomeResource: self.canCreateSomeResource
+          };
+        } else {
+          _.each(this.rules, function(rule) {
+            if (!canCreate && _.indexOf(rule.verbs, "create") !== 1) {
+              self.canCreateSomeResource = true;
+              self.wasCanCreateSomeResourceSet = true;
+              pageRules = {
+                createSomeResource: self.canCreateSomeResource
+              };
+            }
+          });
+        }
+      }
+
+      return pageRules;
+    };
+
+    AuthorizationService.prototype.setForceReload = function(bool) {
+      this.forceReload = bool;
+    };
+
+    AuthorizationService.prototype.getForceReload = function() {
+      return this.forceReload; 
+   };
+
+    AuthorizationService.prototype.clearAll = function() {
+      this.rules = null;
+      this.forceReload = false;
+      this.canCreateSomeResource = false;
+      this.wasCanCreateSomeResourceSet = false;
+      this.projectNameRules = null;
+      $interval.cancel(this.intervalTimer);
+    };
+
+    // Determine whether a single action on a specified resource can be performed by user. The review result will be
+    // added into the $scope.canI variable.
+    // In case a resource shall to be added into the $scope.canI variable under a different unique name(eg. not as a 
+    // resource kind) but a different unique name(eg. namespace), use the 'identifier' parameter.  
+    AuthorizationService.prototype.canI = function(ns, verb, kind, $scope, identifier) {
+      var kindObject = _.find(APIService.availableKinds(true), function(obj) {return obj.kind === kind;});
+      var resource = APIService.kindToResource(kind);
+      var object = {
+        kind:"SubjectAccessReview",
+        apiVersion:"v1",
+        verb: verb,
+        resource: resource,
+        namespace: ns
+      };
+      if (kindObject.group) {
+        object.resourceAPIGroup = kindObject.group;
+      }
+      identifier = identifier || resource;
+      if (!$scope.canI) {
+        $scope.canI = {};
+      }
+      if (!$scope.canI[identifier]) {
+        $scope.canI[identifier] = {};
+      }
+      DataService.create('subjectaccessreviews', null, object, {namespace:ns}).then(
+        function(data) {
+          $scope.canI[identifier][verb] = data.allowed;
+        }, function() {
+          $scope.canI[identifier][verb] = false;
+        });
+    };
+
+    return new AuthorizationService();
```

Comment:
Since this is returning a single instance of the service, probably don't need to go through all the trouble of creating a constructor function and properties on a prototype... the service can just `return { /* props */ }`.  Would eliminate some `var self = this` kind of things as well.  `imagestreams.js` service is prob a decent example.